awx/main/models/events.py

```diff
@@ -538,23 +539,38 @@ def _update_host_summary_from_stats(self, hostnames):
 
             from awx.main.models import Host, JobHostSummary, HostMetric  # circular import
 
-            all_hosts = Host.objects.filter(pk__in=self.host_map.values()).only('id', 'name')
+            if self.job.inventory.kind == 'constructed':
+                all_hosts = Host.objects.filter(id__in=self.job.inventory.hosts.values_list(Cast('instance_id', output_field=models.IntegerField()))).only(
+                    'id', 'name'
+                )
+                constructed_host_map = self.host_map
+                host_map = {}
+                for host in all_hosts:
+                    host_map[host.name] = host.id
```

Comment:
Might as well use a comprehension here.

```suggestion
                host_map = {host.name: host.id for host in all_hosts}
```